src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CB30(CollisionContext* colCtx, CollisionPoly* outPoly, Vec3f* pos) {
+    CollisionPoly* tempPoly;
+    f32 result;
+    s32 bgId;
 
-    if (var1 << 4 < 0) {
-        result = 1;
-    } else {
-        result = 0;
+    result = func_8003C614(NULL, colCtx, COLPOLY_IGNORE_NONE, &tempPoly, &bgId, pos, NULL, 0x1C, 1.0f);
+    if (tempPoly != NULL) {
+        *outPoly = *tempPoly;
     }
+    return result;
+}
 
+// Public raycast toward floor
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CBE8(CollisionContext* colCtx, CollisionPoly* outPoly, s32* bgId, Vec3f* pos) {
+    CollisionPoly* tempPoly;
+    f32 result;
+
+    result = func_8003C614(NULL, colCtx, COLPOLY_IGNORE_NONE, &tempPoly, bgId, pos, NULL, 0x1C, 1.0f);
+    if (tempPoly != NULL) {
+        *outPoly = *tempPoly;
+    }
     return result;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_8004213C.s")
+// Public raycast toward floor
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CCA4(CollisionContext* colCtx, CollisionPoly** outPoly, s32* bgId, Vec3f* pos) {
+    return func_8003C614(NULL, colCtx, COLPOLY_IGNORE_CAMERA, outPoly, bgId, pos, NULL, 0x06, 1.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042244.s")
+// Public raycast toward floor
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CD08(CollisionContext* colCtx, CollisionPoly** outPoly, s32* bgId, Actor* actor, Vec3f* pos) {
+    return func_8003C614(NULL, colCtx, COLPOLY_IGNORE_ENTITY, outPoly, bgId, pos, actor, 0x02, 1.0f);
+}
 
-u8 func_80042538(CollisionContext* colCtx, s32 iParm2) {
-    u32 var1 = *(u32*)(iParm2 + 0xc);
-    return var1;
+// Public raycast toward floor
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CD70(CollisionContext* colCtx, CollisionPoly** outPoly, s32* bgId, Vec3f* pos) {
+    return func_8003C614(NULL, colCtx, COLPOLY_IGNORE_ENTITY, outPoly, bgId, pos, NULL, 0x06, 1.0f);
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042548.s")
+// NTSC 1.0 8002F8E0
+// Tests if moving from `posPrev` to `posNext` will collide with a "wall"
+//`radius` is used to form a sphere for collision detection purposes
+//`checkHeight` is the positive height above posNext to perform certain checks
+// returns true if a collision is detected, else false
+//`outPoly` returns the closest poly detected, while `outBgId` returns the poly owner
+s32 func_8003CDD4(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResult, Vec3f* posNext, Vec3f* posPrev, f32 radius,
+                  CollisionPoly** outPoly, s32* outBgId, Actor* actor, f32 checkHeight, u8 argA) {
+    Lookup* lookupTbl;
+    f32 temp_f0;
+    s32 result;
+    CollisionPoly* poly;
+    f32 dx, dy, dz; // change between posPrev to posNext
```

Comment:
we generally dont do declarations on the same line like this, would separate them out
same with the one toward the bottom of this list